cluster-agent/README.md

```diff
@@ -8,7 +8,44 @@ This chart deploys AppDynamics ClusterAgent for monitoring Kubenetes and OpenShi
 * Kubernetes 1.7+ OpenShift 3.7+
 * Helm and tiller installed in the cluster with the appropriate RBAC settings
 * Kubernetes [Metrics server](https://hub.helm.sh/charts/stable/metrics-server) installed
-* An AppDynamics account. AppDynamics offers a [free trial](https://www.appdynamics.com/free-trial/)
+* An AppDynamics account.
+
+### Create a secret
+Secure and recommended approach to provide sensitive data like passwords is to generate a secret containing the sensitive data and providing the pre-created secret name during chart deployment.
+
+You can provide an existing secret containing `api-user` if applicable `controller-key` and `event-key`.
+The existing secret can be provided as a value to `existingSecret`.
+
+If creating a secret, it is recommended to name the secret as `release-name`-`secret-name`. For example:  MyRelease-MySecret
+
+This allows the secrets to be affiliated with the release, but not directly included with the release.
+Use below command to create a secret:
+```
+kubectl create secret generic my-release-<secret_name> \
+  --from-literal='api-user=XXXX' \
+  --from-literal='controller-key=XXXX' \
+  --from-literal='event-key=XXXX' \
+  --namespace namespace_name
+```
+
+### Image Security Policies
```

Comment:
ICP clusters generally have image security policies enforced. Image repositories have to be whitelisted so that images can be pulled during chart install.